src/node_http_common.h

```diff
@@ -453,6 +453,16 @@ class NgRcBufPointer : public MemoryRetainer {
   bool internalizable_ = false;
 };
 
+template <typename allocator_t>
+struct NgHeaderImpl : public MemoryRetainer {
```

Comment:
Can we find a better name for this? `NgHeaderBase`? `Impl` would imply that it implements an already-defined interface, but it does the opposite here